### PR TITLE
[clang-tidy] Simplify boolean expressions

### PR DESCRIPTION
**Pull request by neheb**
Found with readability-simplify-boolean-expr

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: folly/io/async/AsyncSSLSocket.cpp [hunk 1]
```diff
@@ -1005,14 +1003,14 @@ bool AsyncSSLSocket::willBlock(
...
-  } else if (error == SSL_ERROR_WANT_WRITE) {
+  if (error == SSL_ERROR_WANT_WRITE) {
     VLOG(3) << "AsyncSSLSocket(fd=" << fd_ << ", state=" << int(state_)
             << ", sslState=" << sslState_ << ", events=" << eventFlags_ << "): "
             << "SSL_ERROR_WANT_WRITE";
     // Register for write event if not already.
     updateEventRegistration(EventHandler::WRITE, EventHandler::READ);
     return true;
-  } else if ((false
+  if ((false
```
[the remark this answers is not in the document]
Comment (neheb) [reply folly/io/async/AsyncSSLSocket.cpp]: Hmm? How would these be broken? A return statement precedes them.

File: folly/io/async/AsyncSSLSocket.cpp [hunk 2]
```diff
@@ -76,9 +76,7 @@ inline bool zero_return(int error, int rc, int errno_copy) {
 #ifdef _WIN32
   // on windows underlying TCP socket may error with this code
   // if the sending/receiving client crashes or is killed
-  if (error == SSL_ERROR_SYSCALL && errno_copy == WSAECONNRESET) {
-    return true;
-  }
```
[the remark this answers is not in the document]
Comment (neheb) [reply folly/io/async/AsyncSSLSocket.cpp]: Ehm OK. Reverted.

File: folly/io/async/AsyncSSLSocket.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (Orvid) [reply folly/io/async/AsyncSSLSocket.cpp]: The curly braces are mis-matched.

[the remark this answers is not in the document]
Comment (neheb) [reply folly/io/async/AsyncSSLSocket.cpp]: Fixed.

**Pull request closed without merging**